packages/duckiebot_interface/launch/all_drivers.launch

```diff
@@ -3,64 +3,85 @@
     <arg name="param_file_name" default="default" doc="Specify a param file. ex:default." />
     <arg name="robot_type" default="duckiebot" doc="The type of the robot"/>
 
-    <!-- Joystick: reads from a physical joystick and produces ROS messages -->
-    <include file="$(find joystick)/launch/joystick_node.launch">
-        <arg name="veh" value="$(arg veh)"/>
-        <arg name="param_file_name" value="$(arg param_file_name)"/>
-    </include>
-
     <!-- Camera Node: reads frames from a physical camera and produces ROS messages -->
-    <include file="$(find camera_driver)/launch/camera_node.launch">
-        <arg name="veh" value="$(arg veh)"/>
-        <arg name="param_file_name" value="$(arg param_file_name)"/>
-    </include>
+    <group if="$(eval arg('robot_type') == 'duckiebot')">
+        <include file="$(find camera_driver)/launch/camera_node.launch">
+            <arg name="veh" value="$(arg veh)"/>
+            <arg name="param_file_name" value="$(arg param_file_name)"/>
+        </include>
+    </group>
 
-    <!-- Wheel Drivers: reads ROS messages and spins the motors -->
-    <include file="$(find wheels_driver)/launch/wheels_driver_node.launch">
-        <arg name="veh" value="$(arg veh)"/>
-        <arg name="param_file_name" value="$(arg param_file_name)"/>
-    </include>
+    <!-- Camera Node: for watchtower, has a higher resolution -->
+    <group unless="$(eval arg('robot_type') == 'duckiebot')">
```

Comment:
I think this would be better if it checks for 'watchtower' instead of 'not duckiebot'. In future there might be a third alternative (e.g. traffic light without a camera, duckiebot without a camera).

I would actually drop all the `unless` statements in the file and explicitly specify *when* to use the groups, not *when not to*.